packages/product-feed-luigis-box/src/Model/Setting/LuigisBoxSettingFacade.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ProductFeed\LuigisBoxBundle\Model\Setting;
+
+use Doctrine\ORM\EntityManagerInterface;
+
+class LuigisBoxSettingFacade
+{
+    /**
+     * @param \Shopsys\ProductFeed\LuigisBoxBundle\Model\Setting\LuigisBoxSettingRepository $luigisBoxSettingRepository
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     * @param \Shopsys\ProductFeed\LuigisBoxBundle\Model\Setting\LuigisBoxSettingDataFactory $luigisBoxSettingDataFactory
+     * @param \Shopsys\ProductFeed\LuigisBoxBundle\Model\Setting\LuigisBoxSettingFactory $luigisBoxSettingFactory
+     */
+    public function __construct(
+        protected readonly LuigisBoxSettingRepository $luigisBoxSettingRepository,
+        protected readonly EntityManagerInterface $entityManager,
+        protected readonly LuigisBoxSettingDataFactory $luigisBoxSettingDataFactory,
+        protected readonly LuigisBoxSettingFactory $luigisBoxSettingFactory,
+    ) {
+    }
+
+    /**
+     * @param string $name
+     * @param int $domainId
+     * @param mixed $value
+     */
+    public function saveSetting(string $name, int $domainId, mixed $value): void
+    {
+        $this->luigisBoxSettingRepository->deleteByNameAndDomainId($name, $domainId);
+
+        $lugisBoxSettingData = $this->luigisBoxSettingDataFactory->create();
+        $lugisBoxSettingData->name = $name;
+        $lugisBoxSettingData->domainId = $domainId;
+        $lugisBoxSettingData->value = $value;
+
+        $luigisBoxSetting = $this->luigisBoxSettingFactory->create($lugisBoxSettingData);
+
+        $this->entityManager->persist($luigisBoxSetting);
+        $this->entityManager->flush();
+    }
+
+    /**
+     * @param int $domainId
+     * @return \Shopsys\ProductFeed\LuigisBoxBundle\Model\Setting\LuigisBoxSetting[]
+     */
+    public function getSettingsByDomainId(int $domainId): array
+    {
+        return $this->luigisBoxSettingRepository->getSettingsByDomainId($domainId);
+    }
+
+    public function deleteAll(): void
+    {
+        $this->luigisBoxSettingRepository->deleteAll();
+    }
+
+    /**
+     * @param int $domainId
+     * @return int
+     */
+    public function getLuigisBoxRank(int $domainId): int
```

Comment:
this method is called for each out-of-stock product while generating feed. Shouldn't it be cached, because the value is not expected to be changed in the middle of the feed

Reply:
good idea, afbea52088